homeassistant/components/here_travel_time/sensor.py

```diff
@@ -0,0 +1,285 @@
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_LATITUDE, ATTR_LONGITUDE, LENGTH_METERS, CONF_MODE, CONF_NAME,
+    EVENT_HOMEASSISTANT_START)
+from homeassistant.helpers import location
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.util.dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_DESTINATION = 'destination'
+CONF_ORIGIN = 'origin'
+CONF_APP_ID = 'app_id'
+CONF_APP_CODE = 'app_code'
+CONF_TRAFFIC_MODE = 'traffic_mode'
+CONF_ROUTE_MODE = 'route_mode'
+
+DEFAULT_NAME = "HERE Travel Time"
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=5)
+
+TRAVEL_MODE_CAR = 'car'
+TRAVEL_MODE_PEDESTRIAN = 'pedestrian'
+TRAVEL_MODE_PUBLIC = 'publicTransport'
+TRAVEL_MODE_TRUCK = 'truck'
+TRAVEL_MODE = [
+    TRAVEL_MODE_CAR,
+    TRAVEL_MODE_PEDESTRIAN,
+    TRAVEL_MODE_PUBLIC,
+    TRAVEL_MODE_TRUCK,
+]
+TRAFFIC_MODE_ENABLED = "traffic:enabled"
+TRAFFIC_MODE_DISABLED = "traffic:disabled"
+
+ROUTE_MODE_FASTEST = 'fastest'
+ROUTE_MODE_SHORTEST = 'shortest'
+ROUTE_MODE = [ROUTE_MODE_FASTEST, ROUTE_MODE_SHORTEST]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_APP_ID): cv.string,
+        vol.Required(CONF_APP_CODE): cv.string,
+        vol.Required(CONF_DESTINATION): cv.string,
+        vol.Required(CONF_ORIGIN): cv.string,
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_MODE, default=TRAVEL_MODE_CAR): vol.In(TRAVEL_MODE),
+        vol.Optional(
+            CONF_ROUTE_MODE, default=ROUTE_MODE_FASTEST
+        ): vol.In(ROUTE_MODE),
+        vol.Optional(CONF_TRAFFIC_MODE, default=False): cv.boolean,
+    }
+)
+
+TRACKABLE_DOMAINS = ['device_tracker', 'sensor', 'zone', 'person']
+DATA_KEY = 'here_travel_time'
+
+
+def convert_time_to_utc(timestr):
+    """Take a string like 08:00:00 and convert it to a unix timestamp."""
+    combined = datetime.combine(
+        dt_util.start_of_local_day(), dt_util.parse_time(timestr)
+    )
+    if combined < datetime.now():
+        combined = combined + timedelta(days=1)
+    return dt_util.as_timestamp(combined)
+
+
+def setup_platform(hass, config, add_entities_callback, discovery_info=None):
+    """Set up the HERE travel time platform."""
+    def run_setup(event):
+        """
+        Delay the setup until Home Assistant is fully initialized.
+
+        This allows any entities to be created already
+        """
+        hass.data.setdefault(DATA_KEY, [])
+
+        travel_mode = config.get(CONF_MODE)
+        traffic_mode = config.get(CONF_TRAFFIC_MODE)
+        route_mode = config.get(CONF_ROUTE_MODE)
+
+        name = config.get(CONF_NAME, DEFAULT_NAME)
+        app_id = config.get(CONF_APP_ID)
+        app_code = config.get(CONF_APP_CODE)
+        origin = config.get(CONF_ORIGIN)
+        destination = config.get(CONF_DESTINATION)
+
+        sensor = HERETravelTimeSensor(
```

Comment:
instead of passing so many arguments can we just pass the config object itself?

Reply:
I implemented the same structure as the waze_travel_time component